amethyst_ui/src/bundle.rs

```diff
@@ -24,26 +24,31 @@ use crate::{
 ///
 /// Will fail with error 'No resource with the given id' if the InputBundle is not added.
 #[derive(new)]
-pub struct UiBundle<A = String, B = String, C = NoCustomUi, W = u32, G = ()> {
+pub struct UiBundle<R, I, T, A = String, B = String, C = NoCustomUi, W = u32, G = ()> {
```

Comment:
needs docs lol

Reply:
Yeah, the entire `bundle.rs` and `prefab.rs` need to be updated, there is a metric ton of generics now and I didn't want to update it until I'm through with it

Reply:
Is there no way we can reduce those?

Reply:
@Frizi just did a major refactor to use boxed trait objects instead of generics for this, since it has practically no performance impact and makes this code readable again =P This means that this will continue once he is done with that.